src/ts_catalog/continuous_aggs_watermark.c

```diff
@@ -384,6 +386,18 @@ cagg_watermark_update_internal(int32 mat_hypertable_id, int64 new_watermark, boo
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("watermark not defined for continuous aggregate: %d", mat_hypertable_id)));
 	}
+
+	/*
+	 * We claim that '_timescaledb_functions.cagg_watermark' is IMMUTABLE. However, that is not
+	 * entirely true but needed for constification of the function value and plan-time chunk
+	 * exclusion. The value of the cagg_watermark function changes as soon as we change the
+	 * watermark (so the function value is STABLE, but for this volatility classification, no plan
+	 * time evaluation and chunk exclusion can be performed).
+	 *
+	 * Send an invalidation for the hypertable to invalidate prepared statements on this
+	 * table and force a re-planning using the new watermark.
+	 */
+	CacheInvalidateRelcacheByRelid(ht_relid);
```

Comment:
This invalidation should happen only when `materialized_only=false`

Reply:
@fabriziomello Good catch. Changed.